### PR TITLE
[proto] Added tests for other padding modes

### PR DESCRIPTION
**Pull request by vfdev-5**
Related to https://github.com/pytorch/vision/issues/5782

Description:
- Added missing tests for other padding modes: "edge", "reflect", "symmetric"

### CODE REVIEW

File: test/test_prototype_transforms_functional.py [hunk 1]
```diff
@@ -1168,25 +1157,67 @@ def _compute_expected_bbox(bbox, padding_):
         torch.testing.assert_close(output_boxes, expected_bboxes)
 
 
+@pytest.mark.parametrize("device", cpu_and_gpu())
+def test_correctness_pad_segmentation_mask_on_fixed_input(device):
```
Comment (pmeier) [inline test/test_prototype_transforms_functional.py:1161]: Does this test have any value with the other test added?

File: test/test_prototype_transforms_functional.py [hunk 2]
```diff
@@ -1168,25 +1157,67 @@ def _compute_expected_bbox(bbox, padding_):
...
+            output[..., :pad_up, :pad_left] = mask[..., 0, 0].unsqueeze(-1).unsqueeze(-2)
+            output[..., pad_up:-pad_down, :pad_left] = mask[..., :, 0].unsqueeze(-1)
+            output[..., -pad_down:, :pad_left] = mask[..., -1, 0].unsqueeze(-1).unsqueeze(-2)
+
+            output[..., :pad_up, -pad_right:] = mask[..., 0, -1].unsqueeze(-1).unsqueeze(-2)
+            output[..., pad_up:-pad_down, -pad_right:] = mask[..., :, -1].unsqueeze(-1)
+            output[..., -pad_down:, -pad_right:] = mask[..., -1, -1].unsqueeze(-1).unsqueeze(-2)
+
+            output[..., :pad_up, pad_left:-pad_right] = mask[..., 0, :].unsqueeze(-2)
+            output[..., -pad_down:, pad_left:-pad_right] = mask[..., -1, :].unsqueeze(-2)
```
Comment (pmeier) [inline test/test_prototype_transforms_functional.py:1199]: This is quite hard to parse, but I guess there is little we can do to improve. Two suggestions though:

1. If you specify a range as index, the dimensions will not be squeezed. For example: `mask[..., 0, 0].unsqueeze(-1).unsqueeze(-2) == mask[..., :1, :1]`
2. Could you add a comment above each block which side, i.e. top, bottom, left, or right, the padding applies to? Please also do so in the branch below.

File: test/test_prototype_transforms_functional.py [hunk 3]
```diff
@@ -1168,25 +1157,67 @@ def _compute_expected_bbox(bbox, padding_):
...
-    def _compute_expected_mask(mask, padding_):
+@pytest.mark.parametrize("padding_mode", ["constant", "edge", "reflect", "symmetric"])
+def test_correctness_pad_segmentation_mask(padding, padding_mode):
+    def _compute_expected_mask(mask, padding_, padding_mode_):
         h, w = mask.shape[-2], mask.shape[-1]
         pad_left, pad_up, pad_right, pad_down = _parse_padding(padding_)
 
+        # expected output can be computed on positive pad values only
+        # but F.pad_* can also crop for negative values
+        assert pad_left > 0 and pad_up > 0 and pad_right > 0 and pad_down > 0
```
Comment (pmeier) [inline test/test_prototype_transforms_functional.py:1180]: Let's make it clear that this is not an error of the op.

```suggestion
        if any(pad <= 0 for pad in [pad_left, pad_up, pad_right, pad_down])
            raise pytest.UsageError(
                "expected output can be computed on positive pad values only, "
                "but F.pad_* can also crop for negative values"
            )
```

**Review by pmeier — COMMENTED** [review]
A few minor comments inline.

File: test/test_prototype_transforms_functional.py (hunk 1 above)
Comment (vfdev-5) [reply test/test_prototype_transforms_functional.py:1161]: This test checks one option on cuda device instead of [cuda, cpu] x all_options from below test

File: test/test_prototype_transforms_functional.py (hunk 2 above)
Comment (vfdev-5) [reply test/test_prototype_transforms_functional.py:1199]: Let's keep `unsqueeze`, it is much less obvious to parse with `mask[..., :1, :1]`.

**Review by datumbox — APPROVED** [review]
LGTM, thanks!
